fink_science/cats/processor.py

```diff
@@ -128,66 +121,81 @@ def predict_nn(
 
```

Comment:
As you uploaded a new model, the current test suite is failing. New predictions for the test data set give 89 objects for class 0 (see https://github.com/astrolabsoftware/fink-science/actions/runs/4723736769/jobs/8468894307?pr=282#step:7:532). Can you confirm, and update the corresponding test?

Reply:
Hi Julien, I have to admit that I didn't understand exactly, what I should do..

About the predictions, can I have a sample of the test data? I want to inspect the classifications in more detail. 

In principle, outputting classes between [0, 4] (which should represent the classes SN-like, Fast, Long, Periodic and Non Periodic ) it's not a problem but this time we're returning an array with probabilities, not a single Number indicating the class, this can cause an issue?

Reply:
Hi @and-santos -- sorry I was not clear enough. There is no problem with returning an array of probability instead of the previous behaviour. However, the unit test in the function `predict_nn`, should be changed accordingly. Currently it read:

```python
"""
Examples
    -----------
    >>> from fink_utils.spark.utils import concat_col
    >>> from pyspark.sql import functions as F
    >>> df = spark.read.format('parquet').load(elasticc_alert_sample)
    # Assuming random positions
    >>> df = df.withColumn('cdsxmatch', F.lit('Unknown'))
    >>> df = df.withColumn('roid', F.lit(0))
    # Required alert columns
    >>> what = ['midPointTai', 'psFlux', 'psFluxErr', 'filterName']
    # Use for creating temp name
    >>> prefix = 'c'
    >>> what_prefix = [prefix + i for i in what]
    # Append temp columns with historical + current measurements
    >>> for colname in what:
    ...     df = concat_col(
    ...         df, colname, prefix=prefix,
    ...         current='diaSource', history='prvDiaForcedSources')
    # Perform the fit + classification (default model)
    >>> args = [F.col(i) for i in what_prefix]
    >>> args += [F.col('diaObject.mwebv'), F.col('diaObject.z_final'), F.col('diaObject.z_final_err')]
    >>> args += [F.col('diaObject.hostgal_zphot'), F.col('diaObject.hostgal_zphot_err')]
    >>> df = df.withColumn('preds', predict_nn(*args))
    >>> df = df.withColumn('cbpf_class', F.col('preds').getItem(0).astype('int')) ---> to be changed
    >>> df = df.withColumn('cbpf_max_prob', F.col('preds').getItem(1))  ---> to be changed
    >>> df.filter(df['cbpf_class'] == 0).count()  ---> to be changed
    0  ---> to be changed
    """
```
The last 4 lines should be changed into something meaningful, like checking how many `SN-like` you predict. You can do something like:

```python
...
>>> df = df.withColumn('preds', predict_nn(*args))
>>> df.withColumn('argmax', F.expr('array_position(a, array_max(a)) - 1'))
>>> df.filter(df['argmax'] == 0).count()
??
```
The test data set is at `fink_science/data/alerts/elasticc_sample_seed0.parquet`